### PR TITLE
Forced registration of Objects, ID of object

### PR DESCRIPTION
**Pull request by JuniorJPDJ**

### CODE REVIEW

**Comment by JuniorJPDJ** [comment]
@JuniorJPDJ obczajcie to ktos i zrobta review jak mozecie

File: music_service_async_interface/__init__.py [hunk 1]
```diff
@@ -72,6 +72,15 @@ def __lt__(self, other):
...
     obj: Type["Object"]
+    quality: Type[AudioQuality]
+
+    @abstractmethod
+    def __init__(
+        self, obj_cls: Type["Object"], quality_cls: Type[AudioQuality], sess: Optional[aiohttp.ClientSession] = None
+    ):
+        self.quality = quality_cls
+        self.obj = obj_cls
+        self.sess = aiohttp.ClientSession() if sess is None else sess
```
Comment (wvffle) [inline music_service_async_interface/__init__.py:83]: `self.sess = sess`

File: music_service_async_interface/__init__.py [hunk 2]
```diff
@@ -72,6 +72,15 @@ def __lt__(self, other):
 class Session(ABC):
     sess: aiohttp.ClientSession
     obj: Type["Object"]
+    quality: Type[AudioQuality]
+
+    @abstractmethod
+    def __init__(
+        self, obj_cls: Type["Object"], quality_cls: Type[AudioQuality], sess: Optional[aiohttp.ClientSession] = None
```
Comment (wvffle) [inline music_service_async_interface/__init__.py:79]: `sess: ... = aiohttp.ClientSession()`

**Review by wvffle — CHANGES_REQUESTED** [review]

File: music_service_async_interface/__init__.py (hunk 2 above)
Comment (JuniorJPDJ) [reply music_service_async_interface/__init__.py:79]: No, you can't do this with mutable objects.
Default parameter values are calculated on function definition.
You would create all `Session`s without this argument with the same `aiohttp.ClientSession`s.

**Review by divadsn — APPROVED** [review]
LGTM

**Review by divadsn — APPROVED** [review]

**Comment by JuniorJPDJ** [comment]
gr8, @wvffle blocked it bcs of bad python knowledge and left.. ;X

File: music_service_async_interface/__init__.py (hunk 2 above)
Comment (wvffle) [reply music_service_async_interface/__init__.py:79]: ok, seems weird to me

**Review by wvffle — APPROVED** [review]
Seems like @divadsn is lacking some basic knowledge how pull requests work. No, if you self assign yourself and approve the changes, you still won't be able to merge this request.

**Comment by JuniorJPDJ** [comment]
> Seems like @divadsn is lacking some basic knowledge how pull requests work. No, if you self assign yourself and approve the changes, you still won't be able to merge this request.

I asked him to approve this ;x

**Pull request merged**